backend/app/routes/broadcast/getBroadcasts.js

```diff
@@ -0,0 +1,43 @@
+const to = require('await-to-js').default;
+const Broadcast = require('../../models/Broadcast');
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+
+module.exports = async (req, res, next) => {
+  const page = parseInt(req.query.page, 10);
+  const limit = parseInt(req.query.limit, 10);
+  const startIndex = (page - 1) * limit;
+  const endIndex = page * limit;
+  const results = {};
+  if (!page || !limit) {
+    const error = new ErrorHandler(constants.ERRORS.INPUT, {
+      statusCode: 400,
+      message: 'Input Error: Fetching Failed',
+    });
+    return next(error);
+  }
+  if (startIndex > 0) {
```

Comment:
It shouldn't be this complex

Reply:
it is for sending if previous or next page exists

Reply:
If you are going for that approach then there should a pointer to previous one instead of this
I believe that is the standard approach, open for discussion

Reply:
> it is for sending if previous or next page exists

If you are going for that approach then there should a pointer to previous one instead of this
I believe that is the standard approach, open for discussion

Reply:
look, my approach is simple, I will explain it. 
when the get request is sent, it returns a JSON that contains 3 properties.
one of them is the results array which is always present.
The other 2 are previous & next which may be or may not be present depending on the situation.
So in the front end, we can check whether it returns a previous & next page and show the navigation according to it.